src/resources/views/crud/fields/inc/repeatable_row.blade.php

```diff
@@ -16,6 +21,7 @@
         </button>
         @endif
     </div>
+    @if($field['reorder'] && isset($field['orderColumn']) && $field['orderColumn'] !== false)<input type="hidden" class="order_hidden_input" name="{{$reorder_input_name}}" value="" />@endif
```

Comment:
Here's what I propose, to solve both issues in an understandable way. Let's start from the docs. **This is a `repeatable` feature, called `reorder`**. It extends the previous `reorder` functionality in `repeatable`, to make it possible to reorder not just _visually_, but also update a hidden field, when that up/down arrows are pressed. Correct?

How do you add a hidden `reorder` field to a `repeatable`? Let's look at the "modified docs" for the `repeatable` field (look at the bottom):

```php
[   // repeatable
    'name'  => 'testimonials',
    'label' => 'Testimonials',
    'type'  => 'repeatable',
    'subfields' => [
        [
            'name'    => 'name',
            'type'    => 'text',
            'label'   => 'Name',
            'wrapper' => ['class' => 'form-group col-md-4'],
        ],
        [
            'name'    => 'position',
            'type'    => 'text',
            'label'   => 'Position',
            'wrapper' => ['class' => 'form-group col-md-4'],
        ],
        [
            'name'    => 'company',
            'type'    => 'text',
            'label'   => 'Company',
            'wrapper' => ['class' => 'form-group col-md-4'],
        ],
        [
            'name'  => 'quote',
            'type'  => 'ckeditor',
            'label' => 'Quote',
        ],
    ],

    // optional
    'new_item_label'  => 'Add Group', // customize the text of the button
    'init_rows' => 2, // number of empty rows to be initialized, by default 1
    'min_rows' => 2, // minimum rows allowed, when reached the "delete" buttons will be hidden
    'max_rows' => 2, // maximum rows allowed, when reached the "new item" button will be hidden
    'reorder' => false, // hide up&down arrows next to each row (no reordering)
    'reorder' => true, // show up&down arrows next to each row
    'reorder' => 'order', // show arrows and adds a hidden subfield with that name (value gets updated when rows move)
    'reorder' => ['name' => 'order', 'type' => 'number', 'attributes' => ['data-reorder-input' => true]], // shows arrows and add a visible number subfield, which will be updated when rows move up and down
],
```

When starting from the docs we see that:
- rather than calling it "order" it makes more sense to call it "reorder" because we already have that notion;
- rather than defining a "column" it makes more sense to define a "subfield"; but that would be odd, and we'd have to use underscore to match the other features inside `repeatable`... so `reorder_subfield` which is a bit ugly so... why not use `reorder` itself? in the end, if `reorder => false` the hidden field doesn't make sense anyway;

I think this makes the syntax more consistent with the other features inside `repeatable`, and easy to understand. You can define a few stuff to `reorder`:
- true/false
- a string => hidden input
- an array => that input

This means a dev can do:

```php
CRUD::field('testimonials')->subfields([
    'title',
    ['name' => 'body', 'type' => 'textarea'],
])->reorder('order'); // will add a hidden subfield for it

// or if you want something different about that subfield:

CRUD::field('testimonials')->subfields([
    'title',
    ['name' => 'body', 'type' => 'textarea'],
])->reorder([
    'name' => 'order', 
    'type' => 'text', 
    'attributes' => ['data-reorder-input' => true]
]); // if you want to customize something about it (notice I've made it "text" here for some reason)
```

Or manually (which nobody will do):

```php
CRUD::field('testimonials')->subfields([
    'title',
    ['name' => 'body', 'type' => 'textarea'],
    ['name' => 'order', 'type' => 'hidden', 'attributes' => ['data-reorder-input' => true]],
]);
```

All of the above will work. But most people will just do either `reorder(false)` or `reorder('order')`.

----

How this works is using JS. The `repeatable` field sees inputs with `data-reorder-input`, so it knows (inside its JS) to:
- reorder the items by that field, on field initialization;
- when items are reordered, to update all `input[type=hidden][data-reorder-input=true]` inputs in that `repeatable` with the new order values;

----

So sum up, what I'm proposing:
- [ ] let's use `->reorder()` instead of `->orderColumn()`
- [ ] let's add a real subfield, if given by `reorder`:
     - [ ] string => a hidden field with `data-reorder-input=true`
     - [ ] array => that given field
- [ ] let's move the reordering of the rows to JS (no point in Backpack having any notion of the `$subfield['reorder']`, it can be purely a feature offered by that field type;

--- 

What do you think @pxpm ?

Reply:
@tabacitu it's the same problem as with `subfields/pivotFields`, we cannot support it so down in the chain only, because otherwise we have no "reorder_column" in the `pivot fields` when we want to get the values. 

Bare with me:

You setup in controller the field for the `BelongsToMany` with pivot where you want to have a reorder field:

```php
CRUD::field('postalboxes')->subfields([
            [
                'name' => 'postal_name'
            ], 
        ])->reorder('order_column');
```

From here we will reach the `relationship.blade.php`, and from there we would endup with `entries.blade.php`. Only in `entries` we will setup the order field and add it to subfields:

```php
// from entries.blade.php
switch ($field['relation_type']) {
        case 'MorphToMany':
        case 'BelongsToMany':
            $field['subfields'] = Arr::prepend($field['subfields'], $pivotSelectorField);
            break;
        case 'MorphMany':
        case 'HasMany':
            if(isset($entry)) {
                $field['subfields'] = Arr::prepend($field['subfields'], [
                    'name' => $entry->{$field['name']}()->getLocalKeyName(),
                    'type' => 'hidden',
                ]);
            }
            break;
        default: {
            if(is_string($field['reorder'])) {
                $field['subfields'] = Arr::prepend($field['subfields'], [
                    'name' => $field['reorder'],
                    'type' => 'hidden',
                ]);
            }
        }
        break;
    }
```

When you go to the edit page, running the "crud" functions to get the values from database you do:
- get relationship fields from crud fields (your `postalboxes` field will be returned in this example)
- now get the `subfields` from that relationship and return the values (at this time, **the only subfield is `postal_name`**, because we only setup the ´order_column` field after in the blades files. 

So to sum up:
- I agree with you using only `reorder` key for bool/column definition 👍 
- I agree with you that we could add it as a subfield instead of manually define the hidden input 👍 
- I don't think we can make it work if upstream crud functions are not aware at all of this functionality. 👎 

I've refactored to accomodate some of your concerns but could not remove the crud awareneess for reorder when dealing with certain relationships.

Reply:
Ah, ok, I get it now, thank you @pxpm . Maybe the problem isn't that we don't have `order` in `subfields` by then (because it's added in the view). Maybe the problem is that for subfields... we take each attribute individually. I don't see why we would do that, when we can just take ALL attributes in one go. I'm opening a PR to explain this better.

Reply:
Done. Please see https://github.com/Laravel-Backpack/CRUD/pull/4086/files and if that's feasible, let's do that. My tests show it works the same, but I might be missing something.